src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java

```diff
@@ -254,7 +258,11 @@ public boolean apply(Hardware input) {
       virtualHardwareSection = updateVirtualHardwareSection(virtualHardwareSection, processorPredicate, virtualCpus + " virtual CPU(s)", BigInteger.valueOf(virtualCpus.intValue()));
       Predicate<ResourceAllocationSettingData> memoryPredicate = resourceTypeEquals(ResourceAllocationSettingData.ResourceType.MEMORY);
       virtualHardwareSection = updateVirtualHardwareSection(virtualHardwareSection, memoryPredicate, ram + " MB of memory", BigInteger.valueOf(ram.intValue()));
-
+      if (disk != null) {
+         Predicate<ResourceAllocationSettingData> diskPredicate = resourceTypeEquals(ResourceAllocationSettingData.ResourceType.DISK_DRIVE);
+         Predicate<ResourceAllocationSettingData> elementPredicate = elementNameEquals("Hard disk 1");
+         virtualHardwareSection = updateVirtualHardwareSectionDisk(virtualHardwareSection, Predicates.and(diskPredicate, elementPredicate), BigInteger.valueOf(disk.intValue()));
```

Comment:
[minor] would it be possible to apply all the changes using a single `updateVirtualHardwareSection` instead of adding `updateVirtualHardwareSectionDisk`?